napari/_qt/dialogs/qt_reader_dialog.py

```diff
@@ -241,16 +241,18 @@ def prepare_remaining_readers(
         del readers[plugin_name]
     # if there's no other readers left, raise the exception
     if not readers and error:
-        raise ReaderPluginError(
-            trans._(
-                'Tried to read {path_message} with plugin {plugin}, because it was associated with that file extension/because it is the only plugin capable of reading that path, but it gave an error. Try associating a different plugin or installing a different plugin for this kind of file.',
-                path_message=(
-                    f'[{paths[0]}, ...]' if len(paths) > 1 else paths[0]
-                ),
-                plugin=plugin_name,
+        error_msg = trans._(
+            'Tried to read {path_message} with plugin {plugin}, because it was associated with that file extension/because it is the only plugin capable of reading that path, but it gave an error. Try associating a different plugin or installing a different plugin for this kind of file.',
+            path_message=(
+                f'[{paths[0]}, ...]' if len(paths) > 1 else paths[0]
             ),
+            plugin=plugin_name,
+        )
+        raise ReaderPluginError(
+            error_msg,
             plugin_name,
             paths,
+            getattr(error, 'original_error', None),
```

Comment:
What is the benefit from adding a message to string inside `ReaderPluginError` constructor, instead of here?

Reply:
when we discussed this we were considering using the error's stack trace rather than just the message, but maybe now this can be reverted @DragaDoncila?

Reply:
@Czaki in this case we have two "layers" of `ReaderPluginError` that we need to surface the original error through i.e. `error` here isn't the error we're actually trying to show. 

We [first](https://github.com/napari/napari/blob/main/napari/components/viewer_model.py#L1555) `raise ReaderPluginError from e` in `ViewerModel` and then we raise it here in the GUI handling code. I therefore decided to store an `original` error with the `ReaderPluginError` so that regardless of whether we catch this error elsewhere or just raise it here, we have easy access to that original non-napari-reading-code error.

I could just concatenate the original error to the error message in `ViewerModel` and then "re-concatenate` it here, but this seemed cleaner. Open to suggestions!